### PR TITLE
Add setup logic for AWS tenant clusters

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/4046

Extracts the setup logic from aws-operator. Mainly the code is just copied and I've commented the changes. The updated wiring code in aws-operator is here https://github.com/giantswarm/aws-operator/pull/1144.

Maybe this is too big but I'd like to get feedback on it. Open to suggestions for making it smaller.

### CODE REVIEW

File: aws/config.go [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package aws
```
Comment (rossf7) [inline aws/config.go:1]: These are just taken from aws-operator integration dir.

File: aws/setup.go [hunk 2]
```diff
@@ -0,0 +1,307 @@
...
+	credentialName        = "credential-default"
+	credentialNamespace   = "giantswarm"
+	provider              = "aws"
+)
+
+func Setup(ctx context.Context, m *testing.M, config Config) error {
+	var v int
+	var err error
+	var errors []error
+
```
Comment (rossf7) [inline aws/setup.go:35]: Error handling added here.

File: aws/setup.go [hunk 3]
```diff
@@ -0,0 +1,307 @@
...
+	if config.Guest == nil {
+		return microerror.Maskf(invalidConfigError, "%T.Guest must not be empty", config)
+	}
+	if config.Host == nil {
+		return microerror.Maskf(invalidConfigError, "%T.Host must not be empty", config)
+	}
+
+	vpcPeerID, err := installHostPeerVPC(config)
+	if err != nil {
+		errors = append(errors, err)
```
Comment (rossf7) [inline aws/setup.go:48]: Uses the new approach of having a slice of errors.

File: aws/setup.go [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+)
+
+const (
+	awsOperatorArnKey     = "aws.awsoperator.arn"
+	awsResourceValuesFile = "/tmp/aws-operator-values.yaml"
+	credentialName        = "credential-default"
+	credentialNamespace   = "giantswarm"
+	provider              = "aws"
+)
+
```
Comment (rossf7) [inline aws/setup.go:30]: Only the Setup method changes. The rest of the code is copied from aws-operator. Context is passed in from the operator.

File: aws/setup.go [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+			errors = append(errors, err)
+			v = 1
+		}
+	}
+
+	if v == 0 {
+		v = m.Run()
+	}
+
+	if env.KeepResources() != "true" {
```
Comment (rossf7) [inline aws/setup.go:76]: `os.Getenv` changes replaced by `env` package.

File: aws/setup.go [hunk 6]
```diff
@@ -0,0 +1,307 @@
...
+		}
+	}
+
+	if len(errors) > 0 {
+		return microerror.Mask(errors[0])
+	}
+
+	return nil
+}
+
```
Comment (rossf7) [inline aws/setup.go:97]: Rest of the code comes from aws-operator. There are more changes I'd like to make but I'll do it once the library is used by aws-operator and cluster-operator.

File: aws/setup.go (hunk 4 above)
Comment (kopiczko) [reply aws/setup.go:30]: Cool I wanted to do it in the first place, but I didn't want to introduce too many changes at once 👍

**Review by kopiczko — APPROVED** [review]

File: aws/env/aws.go [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+
+func GuestAWSAccessKeyID() string {
+	return guestAWSAccessKeyID
+}
+
+func GuestAWSAccessKeySecret() string {
+	return guestAWSAccessKeySecret
+}
+
+func GuestAWSAccessKeyToken() string {
```
Comment (fgimenez) [inline aws/env/aws.go:119]: Could we remove the token functions if they are not being used?

**Review by fgimenez — APPROVED** [review]
Nice!

File: aws/env/aws.go (hunk 7 above)
Comment (rossf7) [reply aws/env/aws.go:119]: In the end I've updated the functions to set the session tokens.  As when testing locally with individual IAM accounts you need to pass the session token as well as the access key id and secret.

**Comment by rossf7** [comment]
Picking this up after a break. Going with this but next step is to update the library with any recent changes in aws-operator.

**Pull request merged**